modules/appeals_api/app/controllers/appeals_api/v2/decision_reviews/higher_level_reviews/contestable_issues_controller.rb

```diff
@@ -27,6 +32,32 @@ def index
 
   private
 
+  # rubocop:disable Metrics/MethodLength
+  def validate_receipt_date_header!
+    unless Date.parse(request_headers['X-VA-Receipt-Date']).after?(AMA_ACTIVATION_DATE)
+      error = {
+        title: I18n.t('appeals_api.errors.titles.validation_error'),
+        detail: I18n.t('appeals_api.errors.receipt_date_too_early'),
+        source: {
+          header: 'X-VA-Receipt-Date'
+        },
+        status: '422'
+      }
+      render json: { errors: [error] }, status: :unprocessable_entity
+    end
+  rescue Date::Error # If date cannot be parsed
+    error = {
+      title: I18n.t('appeals_api.errors.titles.validation_error'),
+      detail: 'Receipt date has an invalid format.',
+      source: {
+        header: 'X-VA-Receipt-Date'
+      },
+      status: '422'
+    }
+    render json: { errors: [error] }, status: :unprocessable_entity
+  end
+  # rubocop:enable Metrics/MethodLength
```

Comment:
I realized a bug on the Decision Reviews v2 HLR Contestable Issues endpoint (`/services/appeals/v2/decision_reviews/higher_level_reviews/contestable_issues/{benefit_type}`) that doesn't occur on the general Contestable Issues endpoint (`/services/appeals/v2/decision_reviews/contestable_issues/{decision_review_type}`), since the latter endpoint validates the headers against the schema, but the former does not (inherits from the `AppealsApi::V1::DecisionReviews::BaseContestableIssuesController`, which is older). The bug occurred when an `X-VA-Receipt-Date` header was passed but the contents were an invalid date. The code on `L37` was failing to parse the date, and the endpoint was responding with a `500` error. By rescuing the error, I can return a `422` response instead, which is more appropriate.

I started to go down the path of handling this in a more holistic and clean way (I wasn't sure why this endpoint wasn't validating the headers against the schema and the other one was), but the scope spiraled quickly and I didn't want to risk breaking the API contract for this endpoint. This endpoint will soon be replaced by the Appealable Issues v0 API anyway, so I landed on this simpler approach. I chose not to save the error detail in our `I18n` translation file because I don't want to encourage this pattern going forward. (Headers and params should always be compared against the schema in the future.)